### PR TITLE
Update leaderboard style to heatmap

### PR DESCRIPTION
**Pull request by roman807**
closes https://github.com/kaiko-ai/eva/issues/630

check documentation with (1) run `mkdocs serve`, (2) navigate to http://127.0.0.1:8000/leaderboards/

### CODE REVIEW

File: docs/leaderboards.md [hunk 1]
```diff
@@ -9,20 +9,7 @@ We evaluated the following FMs on the 6 supported WSI-classification tasks. We r
...
-| [Lunit - ViT-S16](https://github.com/lunit-io/benchmark-ssl-pathology/releases/)    | TCGA | 0.77|0.936|0.751|0.905|0.869|0.737|0.625|0.549|
-| [Owkin (Phikon) - iBOT ViT-B16](https://huggingface.co/owkin/phikon)                | TCGA | 0.715|0.942|0.766|0.925|0.879|0.784|0.68|0.554|
-| [UNI - DINOv2 ViT-L16](https://huggingface.co/MahmoodLab/UNI)                  | Mass-100k | 0.797|0.95|0.835|0.939|0.933|0.774|0.67|0.575|
-| [kaiko.ai - DINO ViT-S16](https://github.com/kaiko-ai/towards_large_pathology_fms)  | TCGA | 0.8|0.949|0.831|0.902|0.897|0.77|0.622|0.573|
-| [kaiko.ai - DINO ViT-S8](https://github.com/kaiko-ai/towards_large_pathology_fms)	  | TCGA | 0.825|0.948|0.826|0.887|0.879|0.741|0.677|0.617|
-| [kaiko.ai - DINO ViT-B16](https://github.com/kaiko-ai/towards_large_pathology_fms)  | TCGA | 0.846|0.959|0.839|0.906|0.891|0.753|0.647|0.572|
-| [kaiko.ai - DINO ViT-B8](https://github.com/kaiko-ai/towards_large_pathology_fms)   | TCGA | 0.867|0.952|0.814|0.921|0.939|0.761|0.706|0.661|
-| [kaiko.ai - DINOv2 ViT-L14](https://github.com/kaiko-ai/towards_large_pathology_fms)| TCGA | 0.862|0.935|0.822|0.907|0.941|0.769|0.686|0.599|
+![Screenshot](images/leaderboard.svg)
```
Comment (nkaenzig) [inline docs/leaderboards.md:12]: Looks very nice. Would it make sense to also add the script to generate this plot for reproducibility? (parsing the result .json files of different runs etc). Could add this to `scripts/generate_leaderboard_plot.py` or similar.

Comment (roman807) [reply docs/leaderboards.md:12]: added it

File: scripts/generate_leaderboard_plot.py [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+        "consep": "CoNSeP",
+        "monusac": "MoNuSAC",
+    }
+
+    # load results into data frame:
+    all_scores = []
+    for model in _fm_name_map.keys():
+        scores = []
+        for task in _tasks_to_metric.keys():
+            results_folder = [d for d in os.listdir(f"logs/{task}/{model}") if d.startswith("20")][
```
Comment (nkaenzig) [inline scripts/generate_leaderboard_plot.py:53]: should we make the log directory configurable e.g. through argparse?

File: scripts/generate_leaderboard_plot.py [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+                0
+            ]
+            with open(os.path.join(f"logs/{task}/{model}/{results_folder}/results.json")) as f:
+                d = json.load(f)
+            split = "test" if d["metrics"]["test"] else "val"
+            metric = _tasks_to_metric.get(task)
```
Comment (nkaenzig) [inline scripts/generate_leaderboard_plot.py:59]: We might want to raise an exception here or fall back to a default metric

File: scripts/generate_leaderboard_plot.py [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+        axis="x",
+        which="major",
+        labelsize=10,
+        labelbottom=False,
+        bottom=False,
+        top=False,
+        labeltop=True,
+        rotation=20,
+    )
+    plt.savefig("leaderboard.svg", format="svg", dpi=1200, bbox_inches="tight")
```
Comment (nkaenzig) [inline scripts/generate_leaderboard_plot.py:91]: should we make the output_dir configurable through `argparse`, with a default value of `docs/leaderboard.svg`?

File: scripts/generate_leaderboard_plot.py (hunk 2 above)
Comment (roman807) [reply scripts/generate_leaderboard_plot.py]: done

File: scripts/generate_leaderboard_plot.py (hunk 4 above)
Comment (roman807) [reply scripts/generate_leaderboard_plot.py]: set the default to `docs/images/leaderboard_new.svg` (to not by default overwrite the existing one)

File: tools/generate_leaderboard_plot.py [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+        "crc": "CRC",
+        "mhist": "MHIST",
+        "patch_camelyon": "PCam",
+        "camelyon16": "Camelyon16",
+        "panda": "PANDA",
+        "consep": "CoNSeP",
+        "monusac": "MoNuSAC",
+    }
+
+    # get log_dit from arg parser
```
Comment (nkaenzig) [inline tools/generate_leaderboard_plot.py:49]: typo: log_dit

File: tools/generate_leaderboard_plot.py [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--logs_dir", type=str, default="logs")
+    parser.add_argument("--output_file", type=str, default="docs/images/leaderboard_new.svg")
+    args = parser.parse_args()
+
+    # load existing leaderboard if available:
+    if os.path.isfile("tools/data/leaderboard.csv"):
+        df_existing = pd.read_csv("tools/leaderboard.csv")
+        df_existing = df_existing.set_index("Unnamed: 0", drop=True)
```
Comment (nkaenzig) [inline tools/generate_leaderboard_plot.py:58]: Should we give this column a name? e.g. `model`, then we can replace the `Unnamed: 0` here

File: scripts/generate_leaderboard_plot.py (hunk 4 above)
Comment (nkaenzig) [reply scripts/generate_leaderboard_plot.py]: I think it's fine if we overwrite the existing one (we do the same with `leaderboard.csv` & it's always easy to revert using git)

**Review by nkaenzig — APPROVED** [review]
Looks great! Left a few last small comments.

File: tools/generate_leaderboard_plot.py (hunk 6 above)
Comment (roman807) [reply tools/generate_leaderboard_plot.py]: done

File: scripts/generate_leaderboard_plot.py (hunk 4 above)
Comment (roman807) [reply scripts/generate_leaderboard_plot.py]: done

**Pull request merged**